bxin/ptg2hex/hex_diagnostics.ipynb

```diff
@@ -0,0 +1,607 @@
+   "metadata": {},
+   "source": [
+    "### When the LUT is enabled"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "await hex.cmd_setCompensationMode.set_start(enable=1, timeout=10)\n",
+    "lutMode = await hex.evt_compensationMode.aget(timeout=10)\n",
+    "print(\"compsensation mode enabled?\",lutMode.enabled)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pos = await hex.tel_application.aget(timeout=10.)\n",
+    "print('hex position')\n",
+    "for i in range(6):\n",
+    "    print('%10.2f '%pos.position[i], end= ' ')\n",
+    "print('\\n')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "posC = await hex.evt_compensatedPosition.aget(timeout=10.)\n",
+    "print('Compensated position')\n",
+    "for i in 'xyzuvw':\n",
+    "    print('%10.2f '%getattr(posC, i), end= ' ')\n",
+    "print('\\n')\n",
+    "posU = await hex.evt_uncompensatedPosition.aget(timeout=10.)\n",
+    "print('Uncompensated position')\n",
+    "for i in 'xyzuvw':\n",
+    "    print('%10.2f '%getattr(posU, i), end= ' ')\n",
+    "print('\\n')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "h = await mount.evt_heartbeat.next(flush=True, timeout=5)\n",
+    "print(h.heartbeat)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The inputs to the LUT are currently elevation (from mount) and temperature (mount truss? not implemented yet)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mountAngle = await mount.tel_elevation.aget(timeout=10.)\n",
+    "print(\"mount elevation angle\", mountAngle.angleActual)\n",
+    "elev = mountAngle.angleActual"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "LUTfile = '%s/notebooks/ts_config_mttcs/MTHexapod/v1/default.yaml'%(os.environ[\"HOME\"])\n",
+    "with open(LUTfile, 'r') as stream:\n",
+    "    aa = yaml.safe_load(stream)\n",
+    "if hex.salinfo.index == 1:\n",
+    "    elevCoeff = aa['camera_config']['elevation_coeffs']\n",
+    "    tCoeff = aa['camera_config']['temperature_coeffs']\n",
+    "elif hex.salinfo.index == 2:\n",
+    "    elevCoeff = aa['m2_config']['elevation_coeffs']\n",
+    "    tCoeff = aa['m2_config']['temperature_coeffs']"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pred = []\n",
+    "print('Predicted LUT compensation:')\n",
+    "for i in range(6):\n",
+    "    coeff = elevCoeff[i] #starts with C0\n",
+    "    mypoly = np.poly1d(coeff[::-1]) #poly1d needs C5 first\n",
```

Comment:
You might consider np.polynomial.Polynomial. It takes coefficients in the usual order.